src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -1119,23 +1204,11 @@ export class BundleDataClient {
     });
     start = performance.now();
 
-    // Go through expired deposits in this bundle and now prune those that we have seen a fill for to construct
-    // the list of expired deposits we need to refund in this bundle.
-    expiredBundleDepositHashes.forEach((relayDataHash) => {
-      const { deposit, fill } = v3RelayHashes[relayDataHash];
-      assert(isDefined(deposit), "Deposit should exist in relay hash dictionary.");
-      if (
-        !fill &&
-        isDefined(deposit) // Needed for TSC - we check this above.
-      ) {
-        updateExpiredDepositsV3(expiredDepositsToRefundV3, deposit);
-      }
-    });
-
-    // For all deposits older than this bundle, we need to check if they expired in this bundle and if they did,
-    // whether there was a slow fill created for it in a previous bundle that is now unexecutable and replaced
-    // by a new expired deposit refund.
-    await forEachAsync(Array.from(olderDepositHashes), async (relayDataHash) => {
+    // Add any newly expired deposits to the list of expired deposits to refund.
+    // For these refunds, we need to check whether there was a slow fill created for it in a previous bundle
+    // that is now unexecutable and replaced by a new expired deposit refund.
+    const possibleExpiredDeposits = Array.from(olderDepositHashes).concat(Array.from(expiredBundleDepositHashes));
```

Comment:
We now need to treat `expiredBundleDepositHashes` (deposits in this bundle that expired in this bundle) the same as `olderDepositHashes` (deposits from older bundles that expired in this bundle) due to the presence of pre-fills.  This is because a deposit can now be sent so much after a pre-fill that we can never be sure if a fill was sent for a deposit without checking the fillStatuses on-chain mapping. And we never want to send an expired deposit refund for a deposit that was already filled.

This is kind of nice consolidation actually.

Reply:
Actually I think this is wrong because this would mean for a newly expired deposit submitted in the current bundle, we can create an `unexecutableSlowFill` for it without creating a slow fill leaf